libraries/AP_NavEKF3/AP_NavEKF3.h

```diff
@@ -268,6 +268,20 @@ class NavEKF3 {
     bool getRangeBeaconDebug(int8_t instance, uint8_t &ID, float &rng, float &innov, float &innovVar, float &testRatio, Vector3f &beaconPosNED,
                              float &offsetHigh, float &offsetLow, Vector3f &posNED);
 
+    /*
+     * Writes the measurement from a yaw angle sensor
+     *
+     * yawAngle: Yaw angle of the vehicle relative to true north in radians where a positive angle is
+     * produced by a RH rotation about the Z body axis. The Yaw rotation is the first rotation in a
+     * 321 (ZYX) or a 312 (ZXY) rotation sequence as specified by the 'type' argument.
+     * yawAngleErr is the 1SD accuracy of the yaw angle measurement in radians.
+     * timeStamp_ms: System time in msec when the yaw measurement was taken. This time stamp must include
+     * all measurement lag and transmission delays.
+     * type: An integer specifying Euler rotation order used to define the yaw angle.
+     * type = 1 specifies a 312 (ZXY) rotation order, type = 2 specifies a 321 (ZYX) rotation order.
+    */
+    void writeEulerYawAngle(float yawAngle, float yawAngleErr, uint32_t timeStamp_ms, uint8_t type);
```

Comment:
Is this planned to be used in the future? It's unused now.

Reply:
It was the interface intended for the original functionality which was for a generic external yaw sensor. Tridge is using a NMEA GPS interface for this. I would be happier if the GPS yaw was written to this interface rather than via the GPS read.

Reply:
i don't think we should link external yaw fusion to NMEA GPS inside the EKF. It mean we don't get vertical velocity data even if the receiver supports it.

Reply:
OK, I see the interface is still being used and is called from inside the GPS read function

Reply:
Not this one. The one used in the GPS read function is in `AP_NavEKF3_core`.

I'm fine leaving this in if we are thinking of using it in other places in the future (apparently the Marvelmind beacon system can give an accurate heading value, maybe it could be used from there).